### ISSUE TITLE
Updating to v200.8 or v200.9 causes startup error in init.el

### ISSUE THREAD

**Issue by pklapperich** [syl20bnr/spacemacs#8675]
#### Description :octocat:
Updating to spacemacs v200.8 or v200.9 causes startup error: 

```
Warning (initialization): An error occurred while loading ‘/home/paulk/.emacs.d/init.el’:

Symbol's function definition is void: spacemacs-buffer/display-startup-note

To ensure normal operation, you should investigate and remove the
cause of the error in your initialization file.  Start Emacs with
the ‘--debug-init’ option to view a complete error backtrace.
```

I've done a git bisect and found the issue was introduced by 062a4add, which is the revision I have checked out to write this bug report

Side note: When trying to report from spacemacs, pressing C-c C-c opened a new tab in chrome, but as far as I could tell a new issue wasn't submitted. So I've copy/pasted this from my running emacs.

#### Reproduction guide :beetle:
- Update to v200.8 or newer
- Start Emacs

*Observed behaviour:* :eyes: :broken_heart:
The error listed above is displayed.

*Expected behaviour:* :heart: :smile:
No error should be displayed.

#### System Info :computer:
- OS: gnu/linux
- Emacs: 25.1.1
- Spacemacs: 0.200.7
- Spacemacs branch: nil (rev. 062a4add)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(markdown python vimscript yaml php slack gtags helm emacs-lisp git)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GCONF GSETTINGS NOTIFY ACL GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11

#### Backtrace :paw_prints:
Here's what I get if I run with --debug-init
```
Debugger entered--Lisp error: (void-function spacemacs-buffer/display-startup-note)
  (spacemacs-buffer/display-startup-note)
  (if (not (version<= spacemacs-emacs-min-version emacs-version)) (message (concat "Your version of Emacs (%s) is too old. " "Spacemacs requires Emacs version %s or above.") emacs-version spacemacs-emacs-min-version) (load-file (concat (file-name-directory load-file-name) "core/core-load-paths.el")) (require (quote core-spacemacs)) (spacemacs/init) (spacemacs/maybe-install-dotfile) (configuration-layer/sync) (spacemacs-buffer/display-startup-note) (spacemacs/setup-startup-hook) (require (quote server)) (if (server-running-p) nil (server-start)))
  eval-buffer(#<buffer  *load*> nil "/home/paulk/.emacs.d/init.el" nil t)  ; Reading at buffer position 1306
  load-with-code-conversion("/home/paulk/.emacs.d/init.el" "/home/paulk/.emacs.d/init.el" t t)
  load("/home/paulk/.emacs.d/init" t t)
  #[0 "\205\266	\306=\203\307\310Q\202?	\311=\204\307\312Q\202?\313\307\314\315#\203*\316\202?\313\307\314\317#\203>\320\321\322!D\nB\323\202?\316\324\325\324\211#\210\324=\203e\326\327\330\307\331Q!\"\325\324\211#\210\324=\203d\210\203\247\332!\333\232\203\247\334!\211\335P\336!\203\201\211\202\214\336!\203\213\202\214\314\262\203\245\337\"\203\243\340\341#\210\342\343!\210\266\f?\205\264\314\325\344\324\211#)\262\207" [init-file-user system-type delayed-warnings-list user-init-file inhibit-default-init inhibit-startup-screen ms-dos "~" "/_emacs" windows-nt "/.emacs" directory-files nil "^\\.emacs\\(\\.elc?\\)?$" "~/.emacs" "^_emacs\\(\\.elc?\\)?$" initialization format-message "`_emacs' init file is deprecated, please use `.emacs'" "~/_emacs" t load expand-file-name "init" file-name-as-directory "/.emacs.d" file-name-extension "elc" file-name-sans-extension ".el" file-exists-p file-newer-than-file-p message "Warning: %s is newer than %s" sit-for 1 "default"] 7]()
  command-line()
  normal-top-level()
```

**Comment by syl20bnr** [comment]
How do you update Spacemacs ?
What's the output of `git status` in `~/.emacs.d` after the update ?

Finally can you test with a fresh clone of the repository ?

**Comment by pklapperich** [comment]
Sorry it took so long. I've updated spacemacs both using `git pull` and using "update spacemacs" from the welcome page.

Here's what `git status` shows:
`$ git status
HEAD detached at v0.200.9
You are currently bisecting, started from branch 'bd7ef98e'.
  (use "git bisect reset" to get back to the original branch)

nothing to commit, working tree clean`

I'm pretty sure I used a clean checkout when I was bisecting, but I'll try again to be sure.

**Comment by pklapperich** [comment]
I have an alias... `alias emacs='emacsclient -c -a '\'''\'''`

Using 
```
$ unset emacs
$ mkdir ~/spacemacs-test
$ cd ~/spacemacs-test
$ git clone https://github.com/syl20bnr/spacemacs.git .emacs.d
$ cp ~/.spacemacs ./
$ HOME=$HOME/spacemacs-test emacs```

I'm unable to reproduce the error regardless of the steps I take of upgrading/downgrading between 200.7 and 200.9. Using the alias, I'm still unable to reproduce it so long as I am sure to kill the emacs daemon between upgrade/downgrade. I think when I upgraded to 200.9 I killed my emacsclient but left emacs daemon running, and then had the issue. I haven't confirmed, though. When I have reproduced it in the clean environment I haven't done so reliably.

Nobody else has reported this issue; I think it's user error on my part.
